### PR TITLE
Log upstream service time in envoy access log

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

Logging upstream service time enables us to roughly calculate the
network latency between envoy proxy and upstream service. It will be
calculated as: `%DURATION% - %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%`.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by penkeysuresh — APPROVED** [review]
This looks good with respect to L7 log collector perspective in conjunction with https://github.com/tigera/l7-collector/pull/52/, might need a review from code owners of operator.

**Review by rene-dekker — APPROVED** [review]
lgtm

**Pull request merged**
